### ISSUE TITLE
Benchmark and optimize `/admin/job` with large response

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#1212]
In a specific production instance, the search-job action requires ~10 seconds:

![image](https://github.com/fractal-analytics-platform/fractal-server/assets/3862206/3be8cb43-1eb1-46be-b49c-68fa67429a52)

We should understand what is slowing down the API call (the "large" response body, or the common suboptimal ORM usage) and mitigate if possible.

**Comment by tcompa** [comment]
Further tests show very different results for different network scenarios:
- Everything local
- VPN
- VPN + sshuttle

Let's start by focusing on the backend side.
Current tests consistently suggest that fractal-web spends 4-7 seconds in the "waiting" phase, which should correspond to the API response time. Then we observe strong differences in the "Receiving" time (from 57 ms to 27 s!), which are likely something else.

**Comment by tcompa** [comment]
Current status (with @mfranzon):
1. Remove some jobs from current DB -> DONE
This is related to #1140, and achieved with repeated use of
```
delete from applyworkflow where id=XXX and user_email='x@y.z;
```
2. Remove `task_list` from `WorkflowDump` schema -> TODO
3. Remove `task_list` from `job.workflow_dump` creation in endpoint -> TODO
4. Remove history from job.input_dataset_dump (and output) creation in endpoint -> POSTPONED
5. Make job.log optional in get-jobs endpoints -> POSTPONED
6. Consider pagination -> POSTPONED

**Comment by tcompa** [comment]
> Make job.log optional in get-jobs endpoints

Now addressed in #1218
